src/WebJobs.Script/Diagnostics/HealthChecks/HealthCheckExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics.HealthChecks
+{
+    /// <summary>
+    /// Health check related extension methods.
+    /// </summary>
+    internal static class HealthCheckExtensions
+    {
+        /// <summary>
+        /// Registers all health check services required for the functions host. Should be called
+        /// on the WebHost.
+        /// </summary>
+        /// <param name="builder">The builder to register health checks with.</param>
+        /// <returns>The original builder, for call chaining.</returns>
+        public static IHealthChecksBuilder AddWebJobsScriptHealthChecks(this IHealthChecksBuilder builder)
+        {
+            ArgumentNullException.ThrowIfNull(builder);
+            builder
+                .AddWebHostHealthCheck()
+                .AddScriptHostHealthCheck();
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Adds a health check for the web host lifecycle.
+        /// </summary>
+        /// <param name="builder">The builder to register health checks with.</param>
+        /// <returns>The original builder, for call chaining.</returns>
+        public static IHealthChecksBuilder AddWebHostHealthCheck(this IHealthChecksBuilder builder)
+        {
+            ArgumentNullException.ThrowIfNull(builder);
+            builder.AddCheck<ScriptHostHealthCheck>(
```

Comment:
Did we mean to use `ScriptHostHealthCheck` here? If intentional, can we add a comment why (instead of having a WebHostHealthCheck)